### DISCUSSION TITLE
How to extract beta and eta values as a dataframe?

### DISCUSSION THREAD

**Discussion by iamarunrp** [tvtoglu/predictr#3] [General]
How to extract beta and eta values as a dataframe? results in a tabular form ?

**Comment by tvtoglu** [comment]
Hey there!
The Weibull shape and scale parameter are attributes of your class object. You coul do it like this:

```
# Create an instance called sample_object
failures = [0.4508831,  0.68564703, 0.76826143, 0.88231395, 1.48287253, 1.62876357]
sample_object = Analysis(df=failures, bounds='fb')
sample_object.mle()

# Extract beta and eta
# Beta
beta = sample_object.beta

# Eta
eta = sample_object.eta
```

You can directly add them to a dataframe if you like to.

**Comment by iamarunrp** [comment]
Thank you .

It worked perfectly.

Now I have one more question, I don't want the plot to be shown. How to
hide the weibull plot?

I just need eta and beta values.

Regards
Arun

On Fri, Jul 1, 2022 at 12:13 AM tvtoglu ***@***.***> wrote:

> Hey there!
> The Weibull shape and scale parameter are attributes of your class object.
> You coul do it like this:
>
> `
> Create an instance called sample_object
>
> failures = [0.4508831, 0.68564703, 0.76826143, 0.88231395, 1.48287253,
> 1.62876357]
> sample_object = Analysis(df=failures, bounds='fb')
> sample_object.mle()
> Extract beta and eta Beta
>
> beta = sample_object.beta
> Eta
>
> eta = sample_object.eta
> `
>
> You can directly add them to a dataframe if you like to.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/tvtoglu/predictr/discussions/3#discussioncomment-3058782>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ATEYZSHNL635GTBDL4SUW5LVRXTDRANCNFSM52IOLCIA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Thanks & Regards
Arun Kumar R

**Reply by tvtoglu** [reply]
The default value for the show parameter is False, i.e. the plot shouldn't have been shown. However, try to set the show parameter to False like this:
```
failures = [0.4508831,  0.68564703, 0.76826143, 0.88231395, 1.48287253, 1.62876357]
sample_object = Analysis(df=failures, bounds='fb', show=False)
sample_object.mle()
```
As mentioned before you don't have to set it to False, because it is already default value:
```
failures = [0.4508831,  0.68564703, 0.76826143, 0.88231395, 1.48287253, 1.62876357]
sample_object = Analysis(df=failures, bounds='fb') # no show parameter works as well
sample_object.mle()
```
Best regards

**Comment by iamarunrp** [comment]
Thank you for the clarification. It works now

On Wed, Jul 6, 2022 at 5:01 PM tvtoglu ***@***.***> wrote:

> The default value for the show parameter is False, i.e. the plot should
> have been shown. However, try to set the show parameter to False like this:
>
> failures = [0.4508831,  0.68564703, 0.76826143, 0.88231395, 1.48287253, 1.62876357]
> sample_object = Analysis(df=failures, bounds='fb', show=False)
> sample_object.mle()
>
> As mentioned before you don't have to set it to False, because it is
> already default value:
>
> failures = [0.4508831,  0.68564703, 0.76826143, 0.88231395, 1.48287253, 1.62876357]
> sample_object = Analysis(df=failures, bounds='fb') # no show parameter works as well
> sample_object.mle()
>
> Best regards
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/tvtoglu/predictr/discussions/3#discussioncomment-3091698>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ATEYZSH7AEWS6VF6TSRDB6TVSVVB7ANCNFSM52IOLCIA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Thanks & Regards
Arun Kumar R

**Comment by iamarunrp** [comment]
HI -

one followup question on the same discussion thread, if i want to
calculate confidence bounds for the weibull parameters (eta and beta). is
it possible ?

example fisher bounds ?


On Wed, Jul 6, 2022 at 6:57 PM Arun Kumar ***@***.***> wrote:

> Thank you for the clarification. It works now
>
> On Wed, Jul 6, 2022 at 5:01 PM tvtoglu ***@***.***> wrote:
>
>> The default value for the show parameter is False, i.e. the plot should
>> have been shown. However, try to set the show parameter to False like this:
>>
>> failures = [0.4508831,  0.68564703, 0.76826143, 0.88231395, 1.48287253, 1.62876357]
>> sample_object = Analysis(df=failures, bounds='fb', show=False)
>> sample_object.mle()
>>
>> As mentioned before you don't have to set it to False, because it is
>> already default value:
>>
>> failures = [0.4508831,  0.68564703, 0.76826143, 0.88231395, 1.48287253, 1.62876357]
>> sample_object = Analysis(df=failures, bounds='fb') # no show parameter works as well
>> sample_object.mle()
>>
>> Best regards
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/tvtoglu/predictr/discussions/3#discussioncomment-3091698>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ATEYZSH7AEWS6VF6TSRDB6TVSVVB7ANCNFSM52IOLCIA>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***>
>>
>
>
> --
> Thanks & Regards
> Arun Kumar R
>


-- 
Thanks & Regards
Arun Kumar R

**Comment by tvtoglu** [comment]
Hi Arun,

sorry for the late reply.

Yes, you can do that. 
The inverse Fisher information matrix is callable as an attribute of your instance object (only when you use the MLE for regression).

Let's use our initial example:
`
failures = [0.4508831,  0.68564703, 0.76826143, 0.88231395, 1.48287253, 1.62876357]
sample_object = Analysis(df=failures, bounds='fb', show=True) 
sample_object.mle()`

Calling the inverse of the Fisher information matrix f_inv
#f_inv equals cov_matrix, i.e. [var(beta), covar()],[covar(), var(eta)]
sample_object.f_inv
>>
<img width="304" alt="grafik" src="https://user-images.githubusercontent.com/45008505/216837015-b052c677-1841-423a-90f5-2803873587e8.png">

The first item of this matrix is the variance for beta, the last item is the variance for eta.
Your expected value for beta is
`sample_object.beta`

When you use the MLE, beta and eta are assumed to be normally distributed, i.e. you can use the standard way to calculate bounds for them: This would be the 95% intervall. 
```
lower_beta_bound = sample_object.beta - 1.96 * sample_object.f_inv.item(0)
upper_beta_bound = sample_object.beta + 1.96 * sample_object.f_inv.item(0)
```

I hope this helps.

Best regards
